[skip-ci][ntuple] update RNTuple anchor doc (#15700)
root-project/root: tree/ntuple/v7/doc/specifications.md

When serialized to disk, a 64 bit checksum is appended to the anchor, calculated as the XXH3 hash of
all the (serialized) fields of the anchor object.

Note that, since the anchor is serialized as a "classic" TFile key, all integers in the anchor, as well

> as the checksum, are encoded in **big-endian**, and not little-endian like in the RNTuple payload.

Reviewer: ```suggestion
as the checksum, are encoded in **big-endian**, unlike the RNTuple payload which is encoded in little-endian.
```
